lib/Serialization/ModuleFile.cpp

```diff
@@ -784,7 +792,15 @@ void ModuleFile::lookupClassMember(ImportPath::Access accessPath,
       }
     } else {
       for (auto item : *iter) {
-        auto vd = cast<ValueDecl>(getDecl(item.second));
+        auto declOrError = getDeclChecked(item.second);
+        if (!declOrError) {
+          if (!getContext().LangOpts.EnableDeserializationRecovery)
+            fatal(declOrError.takeError());
+          consumeError(declOrError.takeError());
+          continue;
+        }
```

Comment:
We have this block in... quite a few places now. Should we abstract it into another method? Probably fine to do in a follow up either way.

Reply:
We could use an alternative to `getDeclChecked` that drops the error. It could clean up these call sites a bit and we could use it to apply more precise filtering of the errors, to drop only XRef/implementation-only errors for example.